pkg/sbproxy/reconcile/reconcile_visibilities.go

```diff
@@ -134,33 +156,63 @@ func (r *ReconciliationTask) getPlatformVisibilitiesFromCache() []*platform.Serv
 	return nil
 }
 
-func (r *ReconciliationTask) loadPlatformVisibilities(plans []*types.ServicePlan) ([]*platform.ServiceVisibilityEntity, error) {
+func (r *ReconciliationTask) loadPlatformVisibilitiesByBrokers(brokers []platform.ServiceBroker) ([]*platform.ServiceVisibilityEntity, error) {
 	logger := log.C(r.runContext)
 	logger.Debug("ReconciliationTask getting visibilities from platform")
 
-	platformVisibilities, err := r.platformClient.Visibility().GetVisibilitiesByPlans(r.runContext, plans)
+	visibilities, err := r.platformClient.Visibility().GetVisibilitiesByBrokers(r.runContext, brokers)
 	if err != nil {
 		return nil, err
 	}
-	logger.Debugf("ReconciliationTask successfully retrieved %d visibilities from platform", len(platformVisibilities))
+	logger.Debugf("ReconciliationTask successfully retrieved %d visibilities from platform", len(visibilities))
 
-	return platformVisibilities, nil
+	return visibilities, nil
 }
 
-func (r *ReconciliationTask) getSMPlans() ([]*types.ServicePlan, error) {
-	logger := log.C(r.runContext)
-	logger.Debug("ReconciliationTask getting plans from Service Manager")
+func (r *ReconciliationTask) getSMPlansByBrokersAndOfferings(offerings map[string][]*types.ServiceOffering) (map[string][]*types.ServicePlan, error) {
+	result := make(map[string][]*types.ServicePlan)
+	count := 0
+	for brokerID, sos := range offerings {
+		if len(sos) == 0 {
+			continue
+		}
+		// TODO: Do in multiple go routines? The request query might become very long if there are many offerings in one broker
```

Comment:
I suggest removing this comment and create a separate issue for this

Reply:
https://github.com/Peripli/service-broker-proxy/issues/41